docs/src/advanced-concepts/multi-gpu.rst

```diff
@@ -0,0 +1,27 @@
+Multi-GPU training
```

Comment:
As far as I can read on the internet, the code in this PR would work out of the box for multi-CPU training. One would simply have to swap `device: cuda` to `device: cpu` in our options. However, the speed-up factor for training on GPU in our applications is usually around 10-20, so that, assuming perfect parallelism (which I doubt), one would need 10 or 20 CPU nodes to achieve the same training speed as a single GPU.

Given this, I could test if it works and add a docs section on "multi-CPU" training

Reply:
(Although, as you say, one could split a single node into multiple processes)

Reply:
Let me know if you want this to be done now or in a future PR

Reply:
The point of having multi-cpu training is not so much being faster than GPU, but being able to use the hardware efficiently even when the end user does not have a GPU! 

It would be nice if you could give it a quick try and then if it works add it to the documentation. If it does not work out of the box we can leave it for later.

Reply:
I haven't been able to get it to work with gloo. I'll open an issue for this